receiver/kubeletstatsreceiver/scraper.go

```diff
@@ -128,3 +164,83 @@ func (r *kubletScraper) detailedPVCLabelsSetter() func(rb *metadata.ResourceBuil
 		return nil
 	}
 }
+
+func (r *kubletScraper) node(nodeName string) (kubelet.NodeLimits, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+	if n, ok := r.nodes[nodeName]; ok {
+		return n, nil
+	}
+	return kubelet.NodeLimits{}, fmt.Errorf("node does not exist in store: %s", nodeName)
+}
+
+func (r *kubletScraper) start(_ context.Context, _ component.Host) error {
+	if r.nodeInformer != nil {
+		_, err := r.nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc:    r.handleNodeAdd,
+			UpdateFunc: r.handleNodeUpdate,
+			DeleteFunc: r.handleNodeDelete,
+		})
+		if err != nil {
+			r.logger.Error("error adding event handler to node informer", zap.Error(err))
+		}
+		go r.nodeInformer.Run(r.stopCh)
+	}
+	return nil
+}
+
+func (r *kubletScraper) shutdown(_ context.Context) error {
+	r.logger.Debug("executing close")
+	if r.stopCh != nil {
+		close(r.stopCh)
+	}
+	return nil
+}
+
+func (r *kubletScraper) handleNodeAdd(obj any) {
+	if node, ok := obj.(*v1.Node); ok {
+		r.addOrUpdateNode(node)
+	} else {
+		r.logger.Error("object received was not of type v1.Node", zap.Any("received", obj))
+	}
+}
+
+func (r *kubletScraper) handleNodeUpdate(_, newNode any) {
+	if node, ok := newNode.(*v1.Node); ok {
+		r.addOrUpdateNode(node)
+	} else {
+		r.logger.Error("object received was not of type v1.Node", zap.Any("received", newNode))
+	}
+}
+
+func (r *kubletScraper) handleNodeDelete(obj any) {
+	if node, ok := k8sconfig.IgnoreDeletedFinalStateUnknown(obj).(*v1.Node); ok {
+		r.m.Lock()
+		if n, ok := r.nodes[node.Name]; ok {
+			delete(r.nodes, n.Name)
+		}
+		r.m.Unlock()
+	} else {
+		r.logger.Error("object received was not of type v1.Node", zap.Any("received", obj))
+	}
+}
+
+func (r *kubletScraper) addOrUpdateNode(node *v1.Node) {
+	r.m.Lock()
+	if node.Name != "" {
```

Comment:
Don't need to lock if `node.Name` cannot be changed here. Also please use defer
```suggestion
	if node.Name == "" {
		return
	}
	r.m.Lock()
	defer r.m.Lock()
```